### DISCUSSION TITLE
Troubleshooting performance issues: How to profile my Directus installation?

### DISCUSSION THREAD

**Discussion by flyaruu** [directus/directus#2834] [Archive]
I have an underperforming directus installation, and I'm trying to figure out why.
My dataset is pretty small, my database is running fine and is pretty much idle, so I don't think my database is the problem (but I'd like to know for sure). The whole call takes over a second, while running the query that I'm guessing it is performing runs in about 30ms

Is there a setting to have Directus log every query it performs, so I can see if it matches expectations?

Is there a way to have Directus log the time every query takes?

**Comment by Seijinx** [comment]
Personally, when we started our Node/NextJS project on V8 months ago
(8.1?), the API was so slow (+1s in average for simple query, as you
stated) we ended up making our own APIs using Mysql client for Node.

It's a bummer but we expect a great performance boost in V9 with Laravel
(or Node/Express) implementation.

Maybe the core team has a workaround today, but at the times, in our case,
we had none.

We had some difficulties troubleshooting this and pinpoint the exact source
of problems but it's clear that it's not linked to the database.

Best of luck.

Le lun. 31 août 2020 08:29, Frank Lyaruu <notifications@github.com> a
écrit :

**Reply by flyaruu** [reply]
Thanks man, it feels good not to be alone ;-)

**Reply by rijkvanzanten** [reply]
Hoi Frank! v8 doesn't have any internal logging that can be enabled. v9 (currently in alpha, about to move to beta) allows you to configure the log level (down to trace) so you can log everything that happens under the hood.

> It's a bummer but we expect a great performance boost in V9 with Laravel
(or Node/Express) implementation.

I'm seeing about a ~15-25x performance increase locally, but I haven't done any "official" benchmarks yet. Generally speaking it's around _a lot_ to _significantly_ faster. Feel free to take the alpha of the Node API v9 a spin and let me know what you think 🙂 


https://github.com/directus/next

(Also @flyaruu, lemme know what would be useful logs to you so we can make sure the internal logging supports that kind of tracing)

**Comment by rijkvanzanten** [comment]
Hoi Frank! v8 doesn't have any internal logging that can be enabled. v9 (currently in alpha, about to move to beta) allows you to configure the log level (down to trace) so you can log everything that happens under the hood.

> It's a bummer but we expect a great performance boost in V9 with Laravel
(or Node/Express) implementation.

I'm seeing about a ~15-25x performance increase locally, but I haven't done any "official" benchmarks yet. Generally speaking it's around a lot to significantly faster. Feel free to take the alpha of the Node API v9 a spin and let me know what you think 🙂

https://github.com/directus/next

(Also @flyaruu, lemme know what would be useful logs to you so we can make sure the internal logging supports that kind of tracing)

---

_Posted separately so it can be marked as the answer_

**Reply by flyaruu** [reply]
Thanks for the feedback, I'll definitely have a look. As for useful logs: For this situation a "slow operation" log would be useful, where something is logged when an (part of an) operation takes time more than a certain threshold. The tricky part of performance troubleshooting is that there often is no output to look for.
